openmdao/drivers/pyoptsparse_driver.py

```diff
@@ -501,10 +512,32 @@ def _gradfunc(self, dv_dict, func_dict):
 
         try:
 
+            # Assemble inactive constraints
+            inactives = {}
+            if len(self.active_tols) > 0:
```

Comment:
In the ideal world, this is unneccessary, but we have already seen that iteritems on an empty dict is VERY slow, so it is faster to skip this.